src/Lib/Canvas.vala

```diff
@@ -140,814 +87,187 @@ public class Akira.Lib.Canvas : Goo.Canvas {
-                event_y -= new_delta_y;
-                new_width = fix_size (width - new_delta_x);
-                new_height = fix_size (height - new_delta_y);
-                break;
-            case Nob.TOP_CENTER:
-                update_y = event_y < y + height;
-                if (MIN_SIZE > height - new_delta_y) {
-                   new_delta_y = 0;
-                }
-                new_height = fix_size (height - new_delta_y);
-                selected_item.translate (0, new_delta_y);
-                event_y -= new_delta_y;
-                break;
-            case Nob.TOP_RIGHT:
-                update_x = event_x > x;
-                if (!update_x) {
-                    new_delta_x = 0;
-                }
-                update_y = event_y < y + height;
-                new_width = fix_size (width + new_delta_x);
-                if (!update_y) {
-                    new_delta_y = 0;
-                }
-                if (new_delta_y < height) {
-                    selected_item.translate (0, new_delta_y);
-                    //  debug ("translate: %f,%f", 0, new_delta_y);
-                    event_y -= new_delta_y;
-                    new_height = fix_size (height - new_delta_y);
-                }
-                break;
-            case Nob.RIGHT_CENTER:
-                update_x = event_x > x;
-                if (!update_x) {
-                    new_delta_x = 0;
-                }
-                new_width = fix_size (width + new_delta_x);
-                break;
-            case Nob.BOTTOM_RIGHT:
-                update_x = event_x > x;
-                update_y = event_y > y;
-                new_width = fix_size (width + new_delta_x);
-                new_height = fix_size (height + new_delta_y);
-                break;
-            case Nob.BOTTOM_CENTER:
-                update_y = event_y > y;
-                if (!update_y) {
-                    new_delta_y = 0;
-                }
-                new_height = fix_size (height + new_delta_y);
-                break;
-            case Nob.BOTTOM_LEFT:
-                if (new_delta_x > width) {
-                   new_delta_x = 0;
-                }
-                update_y = event_y > y;
-                update_x = event_x < x + width;
-                if (!update_x) {
-                    new_delta_x = 0;
-                }
-                if (new_delta_y == 0) {
-                    if (delta_y > 0 && update_y) {
-                        new_delta_y = delta_y;
-                    } else {
-                        break;
+            default:
+                if (uppercase_keyval <= Gdk.Key.Z && uppercase_keyval >= Gdk.Key.A) {
+                    // Send to ItemsManager to deal with custom user shape
+                    // hotkey preferences from settings
+                    if (items_manager.set_insert_type_from_key (uppercase_keyval)) {
+                        edit_mode = Akira.Lib.Canvas.EditMode.MODE_INSERT;
                     }
```

Comment:
I don't understand this part, can you explain it?
Also, let's avoid indented conditions.

Reply:
Sure! Since this event handler is for "every" key on the keyboard, we should handle the case in which we don't have a key associated with a shape. So we need to change to insert mode only when the key pressed is alpha and is associated to a particular shape. This is done inside the ItemsManager, since in there we could load the user settings and do the mapping between user key and shapes. 

The function returns true only if there's a shape associated with the key being pressed.

I will put the first condition inside the method call, in order to avoid nested condition.